### ISSUE TITLE
Plus character (+) from query is replaced by space character

### ISSUE THREAD

**Issue by josprimasihombing** [vercel/next.js#38543]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.13.1
      npm: 8.1.2
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.2.3-canary.1
      eslint-config-next: 12.2.2
      react: 18.2.0
      react-dom: 18.2.0

### Describe the Bug

Plus character (+) from query is replaced by space character

### Expected Behavior

Plush character is not replaced.

### Link to reproduction

https://github.com/josprimasihombing/nextjs-query-bug-reproduce

### To Reproduce

After the project is pull locally
- npm install
- npm run dev
- go to [http://localhost:3000/?email=foo+bar@bar.com](http://localhost:3000/?email=foo+bar@bar.com)
- see the log in your terminal

**Comment by thebergamo** [comment]
I would recommend a read on this article on MDN exactly about this problem.
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#preserving_plus_signs

TL;DR

There is no way by default that Next.js would know that your intention is to use "+" instead of a whitespace.
Best would be always use `encodeURIComponent('foo+bar@bar.com')` then if you try to access: http://localhost:3000/?email=foo%2Bbar%40bar.com your email would be provided correctly.

**Comment by josprimasihombing** [comment]
Oke, My issue have been solved.
Thanks for your help @thebergamo .
